vimeo/VIMNetworking
# VIM-4074: VOD Model

#### Ticket
**Required for Vimeans only**
[VIM-4074](https://vimean.atlassian.net/browse/VIM-4074)

#### Ticket Summary
Add our model representation for VOD items (films or series)
Augment our representation of video responses to handle videos associated with VOD items (trailer, film, extra content video, or an episode of a series)

#### Implementation Summary
`VIMJSONResponseSerializer`
- added to acceptableContentTypes so we can receive JSON data for ondemand endpoints 

`VIMVODItem`
- these are the kind of objects received for a user's VOD library, films or series they have paid for

`VIMVideo+VOD`
- category on `VIMVideo` to house all VOD related helpers 
- helpers to identify if a video is VOD when your origin is not the user's library, e.g. search results, feed

`VIMConnection` 
- now supports `ondemand` and `trailer` connections
- `videos` connection gives us the the videos for a series 
- new video count properties (e.g. extra, main, viewable counts) describe the number of videos associated with a VOD item

`VIMInteraction`
- buy, rent, subscribe interactions 
- with corresponding properties like stream, expiration and purchase dates 
- this tells us what kind of access someone has on a VOD item or one of its videos (if it was bought, rented or subscribed to)

`VIMVideoPlayRepresentation` 
- corrected some naming inconsistency for `VIMVideoPlayabilityStatus`

Some to-do's 
- API will update the date string format used for `expires_time` and `purchase_time` to match the format we already support. For now we use a helper method to parse this one-off format. See `VIMInteraction` implementation file for details. 
- API will update purchase interactions so they exist on a film object. Purchase logic is in line with anticipated behavior but should be tested. See `VIMVideo+VOD` implementation file for details. 

#### How to Test
Code review only. [See parent PR for more context] (https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/839)  

If helpful, API notes on VOD https://vimean.atlassian.net/wiki/display/DEV/API
